modules/oe_theme_inpage_navigation/tests/src/Kernel/Plugin/Condition/InPageNavigationStateTest.php

```diff
@@ -0,0 +1,220 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_theme_inpage_navigation\Kernel\Plugin\Condition;
+
+use Drupal\Component\Render\FormattableMarkup;
+use Drupal\Core\Plugin\Context\EntityContext;
+use Drupal\oe_theme_inpage_navigation\InPageNavigationHelper;
+use Drupal\Tests\node\Traits\ContentTypeCreationTrait;
+use Drupal\Tests\node\Traits\NodeCreationTrait;
+use Drupal\Tests\oe_theme\Kernel\AbstractKernelTestBase;
+
+/**
+ * Test InPageNavigationState plugin.
+ */
+class InPageNavigationStateTest extends AbstractKernelTestBase {
+
+  use ContentTypeCreationTrait;
+  use NodeCreationTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'filter',
+    'text',
+    'emr',
+    'emr_node',
+    'node',
+    'field',
+    'entity_reference_revisions',
+    'oe_theme_inpage_navigation',
+  ];
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp() {
+    parent::setUp();
+    $this->installEntitySchema('node');
+    $this->installEntitySchema('entity_meta');
+    $this->installEntitySchema('entity_meta_relation');
+    $this->installSchema('node', ['node_access']);
+
+    $this->installConfig([
+      'node',
+      'filter',
+      'emr',
+      'emr_node',
+      'oe_theme_inpage_navigation',
+    ]);
+    $this->createContentType(['type' => 'example', 'name' => 'Example']);
+    $this->container->get('emr.installer')->installEntityMetaTypeOnContentEntityType('oe_theme_inpage_navigation', 'node', ['example']);
+  }
+
+  /**
+   * Tests the inpage navigation state condition.
+   */
+  public function testCondition(): void {
+    /** @var $condition \Drupal\Core\Condition\ConditionPluginBase */
+    $condition = $this->container->get('plugin.manager.condition')->createInstance('oe_theme_inpage_navigation_state');
+    $node = $this->createNode(['type' => 'example', 'title' => 'some title']);
+    foreach ($this->providerTestCondition() as $test_case_name => $test_case) {
+      $condition_instance = clone $condition;
+      $node_instance = clone $node;
```

Comment:
You can just create the node here and on line 81 delete the node instead of the meta. the meta gets deleted two with it.

Reply:
Updated